src/app/Scenes/HomeView/Sections/AuctionResultsRailHomeViewSection.tsx

```diff
@@ -0,0 +1,77 @@
+import { Flex, useScreenDimensions } from "@artsy/palette-mobile"
+import { AuctionResultsRailHomeViewSection_section$key } from "__generated__/AuctionResultsRailHomeViewSection_section.graphql"
+import { BrowseMoreRailCard } from "app/Components/BrowseMoreRailCard"
+import { AuctionResultListItemFragmentContainer } from "app/Components/Lists/AuctionResultListItem"
+import { SectionTitle } from "app/Components/SectionTitle"
+import { extractNodes } from "app/utils/extractNodes"
+import { FlatList } from "react-native"
+import { graphql, useFragment } from "react-relay"
+
+interface AuctionResultsRailHomeViewSectionProps {
+  section: AuctionResultsRailHomeViewSection_section$key
+}
+
+export const AuctionResultsRailHomeViewSection: React.FC<AuctionResultsRailHomeViewSectionProps> = (
+  props
+) => {
+  const section = useFragment(sectionFragment, props.section)
+  const { width: screenWidth } = useScreenDimensions()
+
+  if (!section || !section.auctionResultsConnection?.edges?.length) {
+    return null
+  }
+
+  const auctionResults = extractNodes(section.auctionResultsConnection)
+
+  return (
+    <Flex>
+      <Flex px={2}>
+        <SectionTitle
+          title={section.component?.title ?? "Auction Results"}
+          onPress={() => {
+            // TODO: navigate to Auction Results screen using section.component?.href
+          }}
+        />
+      </Flex>
+      <FlatList
+        horizontal
+        data={auctionResults}
+        showsHorizontalScrollIndicator={false}
+        initialNumToRender={3}
+        renderItem={({ item }) => (
+          <AuctionResultListItemFragmentContainer
+            showArtistName
+            auctionResult={item}
+            width={screenWidth * 0.9}
+          />
+        )}
+        ListFooterComponent={
+          <BrowseMoreRailCard
+            onPress={() => {
+              // TODO: navigate to Auction Results screen using section.component?.behaviors.viewAll.href
+            }}
+            text={section.component?.behaviors?.viewAll?.buttonText ?? "Browse All Results"}
+          />
+        }
+      />
+    </Flex>
+  )
+}
+
+const sectionFragment = graphql`
+  fragment AuctionResultsRailHomeViewSection_section on AuctionResultsRailHomeViewSection {
+    component {
+      title
+      href
+      behaviors {
+        viewAll {
+          buttonText
+          href
+        }
+      }
+    }
+    auctionResultsConnection(first: 10) {
+      ...AuctionResultsRail_auctionResults @relay(mask: false)
```

Comment:
Can we use `totalCount` instead of unmasking the fragment? [The Relay docs recommend against using it.](https://relay.dev/docs/api-reference/graphql-and-directives/#relaymask-boolean) 

```suggestion
      totalCount
      ...AuctionResultsRail_auctionResults
```

Reply:
Good thought about the `totalCount`. I will make that change

Reply:
When it comes to unmasking though, I am not doing it in order to get access to the auction results count, that was a nice side effect to have 😄. I did it in order to solve ts and better introspection.

https://github.com/user-attachments/assets/5a3d3eac-e29a-4191-9da4-5410f223e8ec

Reply:
I think there might be a different way though to work around this. Thanks for the comment

Reply:
I think TypeScript is correct. If you don't unmask, you cannot access the connection-related fields in `auctionResultsConnection` and not pass it to `extractNodes`. You should only be able to read `totalCount`.

Reply:
I needed actually also the auctionResults that's why I updated my approach [here](https://github.com/artsy/eigen/pull/10662/commits/618218f5928c6e422723971d5e8dfd1c99f4f462)